File: tagalong/010-switch/switch.py
import datetime
import time

today = datetime.date.today()

# This control structure is poorly supported
# in VSCode as of October 2022, Python 3.10. Quite hard to write.
match today.weekday():
    case 5:
        print("Today.")
    case 4:
        print("Tomorrow.")
    case 5:
        print("In two days.")
    case _:
        print("Too far away.")

File: tagalong/010-switch/switch.go
package main

import (
	"fmt"
	"time"
)

func main() {
	fmt.Println("When's Saturday?")
	today := time.Now().Weekday()
	switch today {
	case time.Saturday:
		fmt.Println("Today.")
	case time.Friday:
		fmt.Println("Tomorrow.")
	case time.Thursday:
		fmt.Println("In two days.")
	default:
		fmt.Println("Too far away.")
	}
}
